File: app/src/main/java/ru/melod1n/schedule/adapter/ScheduleAdapter.java
package ru.melod1n.schedule.adapter;


import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Locale;
import java.util.Random;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.melod1n.schedule.R;
import ru.melod1n.schedule.common.ThemeEngine;
import ru.melod1n.schedule.current.BaseAdapter;
import ru.melod1n.schedule.current.BaseHolder;
import ru.melod1n.schedule.items.Lesson;
import ru.melod1n.schedule.items.Location;
import ru.melod1n.schedule.items.Subject;

public class ScheduleAdapter extends BaseAdapter<Lesson, ScheduleAdapter.ViewHolder> {

    private int[] colors;

    public ScheduleAdapter(Context context, ArrayList<Lesson> items) {
        super(context, items);
        initColors();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup vg, int i) {
        View v = getInflater().inflate(R.layout.fragment_schedule_item, vg, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NotNull @NonNull ScheduleAdapter.ViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        holder.bind(position);
    }

    @Override
    public void destroy() {

    }

    private String getType(int i) {
        return i == 0 ? "Лекция" : i == 1 ? "Практика" : i == 2 ? "Лабораторная" : "Дополнительное";
    }

    public void initColors() {
        colors = !ThemeEngine.isDark() ? ThemeEngine.COLOR_PALETTE_DARK : ThemeEngine.COLOR_PALETTE_LIGHT;
    }

    class ViewHolder extends BaseHolder {

        @BindView(R.id.lessonType)
        TextView lessonType;

        @BindView(R.id.lessonName)
        TextView lessonName;

        @BindView(R.id.lessonClassroom)
        TextView lessonClassroom;

        @BindView(R.id.lessonStartTime)
        TextView lessonStartTime;

        @BindView(R.id.lessonEndTime)
        TextView lessonEndTime;

        @BindView(R.id.lessonLine)
        View lessonLine;

//        @BindView(R.id.lessonTeacher)
//        TextView lessonTeacher;

        ViewHolder(View v) {
            super(v);
            ButterKnife.bind(this, v);

        }

        @Override
        public void bind(final int position) {
            Lesson item = getItem(position);

            lessonType.setText(String.format(Locale.getDefault(), "%d: %s", position + 1, getType(new Random().nextInt(4))));

            Subject subject = item.getSubject();

            assert subject != null;

            lessonName.setText(subject.getTitle());

            Location location = item.getClassRoom();

            assert location != null;

            lessonClassroom.setText(String.format("%s, %s", location.getTitle(), location.getBuilding()));

            int color = colors[new Random().nextInt(colors.length - 1)];

            lessonLine.setBackgroundColor(color);
            lessonType.setTextColor(color);

            lessonStartTime.setText("8:00");
            lessonEndTime.setText("8:40");
        }
    }
}

File: app/src/main/java/ru/melod1n/schedule/adapter/ScheduleAdapter.kt
package ru.melod1n.schedule.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import ru.melod1n.schedule.R
import ru.melod1n.schedule.base.BaseAdapter
import ru.melod1n.schedule.base.BaseHolder
import ru.melod1n.schedule.model.ScheduleItem
import java.util.*

class ScheduleAdapter(context: Context, items: ArrayList<ScheduleItem>) : BaseAdapter<ScheduleItem, ScheduleAdapter.ViewHolder>(context, items) {

    override fun onCreateViewHolder(parent: ViewGroup, i: Int): ViewHolder {
        return ViewHolder(view(R.layout.fragment_schedule_item, parent))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.bind(position)
    }

    override fun destroy() {}

    private fun getType(i: Int): String {
        return if (i == 0) "Лекция" else if (i == 1) "Практика" else if (i == 2) "Лабораторная" else "Дополнительное"
    }

    inner class ViewHolder(v: View) : BaseHolder(v) {
        private var type: TextView = v.findViewById(R.id.lessonType)
        private var name: TextView = v.findViewById(R.id.lessonName)
        private var classroom: TextView = v.findViewById(R.id.lessonClassroom)
        private var startTime: TextView = v.findViewById(R.id.lessonStartTime)
        private var endTime: TextView = v.findViewById(R.id.lessonEndTime)
        private var line: View = v.findViewById(R.id.lessonLine)

        override fun bind(position: Int) {
            val item = getItem(position).lesson

            type.text = String.format(Locale.getDefault(), "%d: %s", position + 1, getType(Random().nextInt(4)))

            item.subject?.let {
                name.text = it.title
            }

            item.classroom?.let {
                classroom.text = String.format("%s, %s", it.title, it.building)
            }


            this.startTime.text = "8:00"
            this.endTime.text = "8:40"
        }
    }
}